docs/creating-nodes/testing.md

```diff
@@ -0,0 +1,271 @@
+Particularly in the case of unit testing, there is a good chance that the test code will be executed in different environments, such as that of another developer or a CI/CD environment.
+For this reason, we recommend that you create test code that does not depend on the environment.
+When the time comes to conduct integration testing or system testing, use the actual external system in the test instead of the stub.
+
+#### Creating a stub
+
+By way of example, this section explains how to create a stub for a custom system (external service) that has an HTTP API.
+
+* Using [nock](https://github.com/nock/nock) to create a stub (recommended)  
+
+  `nock` is a module created to test HTTP requests.
+  It implements a request stub by partially overriding the standard `http` module.
+  `nock` allows you to create stubs with relatively simple coding and is easier to learn than `express`.
+
+  * Examples of test code creation
+    * [node-red-web-nodes test code](https://github.com/node-red/node-red-web-nodes/tree/master/test)
+      The test code for most nodes is created by using `nock`.
+
+* Using [express](https://github.com/expressjs/express) to create a stub
+
+  `express` is a framework for web application development.
+  You can use `express` to create a web server deployed locally, and send requests to that web server during testing.
+  The web server works as a stub web server, which makes the test non-dependent on external factors.
+  Consider creating a stub web server in `express` when testing processing that includes special requests beyond the capabilities of `nock`.
+
+  * Example of test code creation
+    * [httprequest node](https://github.com/node-red/node-red/blob/master/test/nodes/core/io/21-httprequest_spec.js)
+      By creating a `testServer` in the test code that returns a specific response and issuing any requests to `testServer`, you can perform testing without connecting to a web server outside the local environment.
+
+### Notes on automatic testing by using test code
+
+* Using a test double  
+  The use of a test double offers many advantages, including freedom from dependencies and the ease with which behavior can be verified.
+  Make sure that you have a thorough understanding of `nock` or `sinon` and are able to use it as the situation requires.
+
+* Using settings.js during testing
+  As of Version 0.1.7, Node Test Helper loads a substantially empty settings.js file that it creates internally.
+  You cannot use a customized settings.js file during testing.
+  If the node you are testing contains code that references settings in settings.js, you might be unable to test it using test code.
+
+* OS dependence  
+  Node-RED runs on the node.js platform and is generally OS-independent provided that node.js is available.
+  However, when executing functions that use OS resources such as file access or command execution, the behavior might differ depending on the operating system.
```

Comment:
This is particularly true when using online workflow pipelines - especially when testing things like timeouts.

Reply:
I added this information in the commit, https://github.com/node-red/node-red.github.io/pull/109/commits/0a963451b35b11f238e03a55584171bc5f7cd676.